src/_pytest/assertion/util.py

```diff
@@ -168,22 +176,47 @@ def assertrepr_compare(config, op: str, left: Any, right: Any) -> Optional[List[
     return [summary] + explanation
 
 
-def _compare_eq_any(left: Any, right: Any, verbose: int = 0) -> List[str]:
-    explanation = []  # type: List[str]
+def _compare_eq_any(
+    left: Any, right: Any, indent_level: int, verbose: int = 0
+) -> List[str]:
+    """
+    Generic comparison function, dispatches to other comparison functions based on the inputs' type.
+    :param left: 1st object passed for comparison.
+    :param right: 2nd object passed for comparison.
+    :param indent_level: indentation level, for better visual perception of the output.
+    :param verbose: decides whether to print full error report or only a part of it.
+    """
+    if not isinstance(indent_level, int):
```

Comment:
I suggest removing this, usually we don't type-check private functions, we assume the caller passes the correct types (or that mypy would catch the mistake).